### ISSUE TITLE
<BR> tags and screen readers

### ISSUE THREAD

**Issue by oliverfoster** [adaptlearning/jquery.a11y#5]
Separation of text is not something that can be avoided on iOS as the screenreaders cursor defines separate readable blocks per tag. PC's do not perform like this and so navigating through text does not usually follow the flow of content. Making text blocks tabbable on a PC solves this problem and makes them behave exactly like the iOS screen readers allowing us to present content to screen readers in the order in which it was intended.

The idea of jquery.a11y is to allow each distinct text block to be tabbable so as to be read separately from one another in a uniform and simple way. The question is, how to define a distinct text block?

There is currently issue with br tags whereby they stop a screen reader when in  "text&lt;br&gt;text" format. So this reads as "text" "text". These two texts should be independantly tabbable as if they were &lt;p&gt;text&lt;/p&gt;&lt;p&gt;text&lt;/p&gt;

The two texts either side of the br should be wrapped for the screen reader to digest.
such that:
"text&lt;br&gt;text" 
becomes 
"&lt;span tabindex="0" &gt;text&lt;/span&gt;&lt;br&gt;&lt;span  tabindex="0" &gt;text&lt;/span&gt;" 
allowing the user to uniformly tab from one text element to the next.

When &lt;br&gt; tags are used with other tags they are not such an issue as they can be easily ignored.
"&lt;span&gt;text&lt;/span&gt;&lt;br&gt;&lt;span&gt;text&lt;/span&gt;" 

When being used alongside styling tags, the behaviour needs deeper consideration: 
"&lt;span&gt;text&lt;/span&gt;&lt;br&gt;&lt;strong&gt;text&lt;/strong&gt;&lt;span&gt;text&lt;/span&gt;" 
whereby the styling tag will appear to screen readers as a distinct and separate text in it's own right as will the two span tags.

The text below will on some screen readers be read as a contiguous string and on others (ipad so far) be considered three separate text blocks.
"&lt;span&gt;text&lt;strong&gt;text&lt;/strong&gt;text&lt;/span&gt;"

This begs the question as what to do with:
"text&lt;br&gt;&lt;strong&gt;text&lt;/strong&gt;&lt;span&gt;text&lt;/span&gt;" 

Or:
"&lt;span&gt;&lt;strong&gt;text&lt;/strong&gt;text&lt;/span&gt;text&lt;br&gt;&lt;strong&gt;text&lt;/strong&gt;&lt;span&gt;text&lt;/span&gt;" 

I am almost finished implementing the algorithm which performs this task seemlessly. If you wish to help please look at:

https://github.com/cgkineo/jquery.a11y/blob/master/jquery.a11y.js#L130-L234

and 

https://github.com/cgkineo/jquery.a11y/blob/master/jquery.a11y.js#L5-L6
